crates/node/src/command.rs

```diff
@@ -210,59 +208,74 @@ pub fn run() -> sc_cli::Result<()> {
-                    } else {
-                        utils::fetch_from_url(
-                            constants::GENESIS_ASSETS_URL.to_string() + file,
-                            madara_path.clone() + "/genesis-assets",
-                        )?;
-                    }
-                }
+            let local_path = utils::get_project_path();
+
+            if let Some(configs_url) = cli.run.configs_url.clone() {
+                utils::fetch_from_url(configs_url, madara_path.clone() + "/configs", cli.run.update_configs)?;
+            } else if let Ok(ref src_path) = local_path {
+                let src_path = src_path.clone() + "/configs/index.json";
+                utils::copy_from_filesystem(src_path, madara_path.clone() + "/configs", cli.run.update_configs)?;
+            } else if !cli.run.disable_url_fetch {
+                utils::fetch_from_url(
+                    constants::DEFAULT_CONFIGS_URL.to_string(),
+                    madara_path.clone() + "/configs",
+                    cli.run.update_configs,
+                )?;
             }
 
-            // TODO confirm with the CI that we are fetching all and fetch dynamically
-            // Issue #982
-            for file in constants::CAIRO_CONTRACTS_FILES {
-                let src_path = utils::get_project_path();
-                if let Ok(src_path) = src_path {
-                    let src_path = src_path + "/configs/cairo-contracts/" + file;
-                    utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
-                } else {
-                    utils::fetch_from_url(
-                        constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                        madara_path.clone() + "/cairo-contracts",
+            let madara_configs: configs::Configs =
+                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+                    .expect("Failed to load madara configs");
+
+            for asset in madara_configs.genesis_assets {
+                if let Ok(ref src_path) = local_path {
+                    let src_path = src_path.clone() + "/configs/genesis-assets/" + &asset.name;
+                    utils::copy_from_filesystem(
+                        src_path,
+                        madara_path.clone() + "/configs/genesis-assets",
+                        cli.run.update_configs,
+                    )?;
+                } else if !cli.run.disable_url_fetch {
+                    configs::fetch_and_validate_file(
+                        madara_configs.remote_base_path.clone(),
+                        asset,
+                        madara_path.clone() + "/configs/genesis-assets/",
+                        cli.run.update_configs,
                     )?;
                 }
             }
 
             if let (Some(chain_spec_url), None) = (cli.run.chain_spec_url.clone(), cli.run.testnet) {
```

Comment:
What if `cli.run.disable_url_fetch` is true at this point? Do we still fetch from URL?

Reply:
if the `disable_url_fetch` is enabled, but we set manually an URL, it should be fetched. E.g.
`--disable-url-fetch --chain-spec-url https://domain.com/chain-spec.json`